src/fragment-components/a-radiobutton.tsx

```diff
@@ -0,0 +1,145 @@
+	componentObj,
+	selected,
+	...rest
+}: any) => {
+	const [fragment, setFragment] = useFragment();
+	const parentComponent = getParentComponent(fragment.data, componentObj);
+
+	const addRadio = (offset = 0) => setFragment({
+		...fragment,
+		data: updatedState(
+			fragment.data,
+			{
+				type: 'insert',
+				component: {
+					type: 'radioButton',
+					value: `${componentObj.id}`,
+					labelText: 'New Option',
+					disabled: false
+				}
+			},
+			parentComponent.id,
+			parentComponent.items.indexOf(componentObj) + offset
+		)
+	});
+	componentObj.value = `${componentObj.id}`;
+
+	return (
+		<Adder
+		active={selected}
+		addButtonsCss={addButtonCss}
+		key={componentObj.id}
+		bottomAction={() => addRadio(1)}>
+			<AComponent
+			selected={selected}
+			headingCss={css`width: fit-content; min-width: 9rem;`}
+			componentObj={componentObj}
+			{...rest}> 
+				<RadioButton
+					id={componentObj.id}
+					name={componentObj.codeContext?.name}
+					labelText={componentObj.labelText}
+					defaultChecked={componentObj.defaultChecked}
+					checked={componentObj.defaultChecked}
+					value= {componentObj.value}
+					disabled= {componentObj.disabled} />
+			</AComponent>
+		</Adder>
+	);
+};
+
+export const componentInfo: ComponentInfo = {
+	component: ARadioButton,
+	styleUI: ARadioButtonStyleUI,
+	render: ({ componentObj, select, remove, selected }) => <ARadioButton
+	componentObj={componentObj}
+	select={select}
+	remove={remove}
+	selected={selected}>
+		{componentObj.labelText}
+	</ARadioButton>,
+	keywords: ['radio','button'],
+	name: 'Radio button',
+	defaultComponentObj: {
+		type: 'radioButton'
+	},
+	image: image,
+	hideFromElementsPane: true,
+	codeExport: {
+		angular: {
+			inputs: ({json}) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Label = "${json.labelText}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Disabled = ${json.disabled};
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Id = "${json.codeContext?.name}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Checked = ${json.defaultChecked};`,
+			outputs: (_) => '',
+			imports: [],
+			code: ({json }) => {
+				return `<ibm-radio
+					[id]="${nameStringToVariableString(json.codeContext?.name)}Id"
+					[value]="${nameStringToVariableString(json.codeContext?.name)}Value"
+					[checked]="${nameStringToVariableString(json.codeContext?.name)}Checked"
+					[disabled]="${nameStringToVariableString(json.codeContext?.name)}Disabled"
+					${angularClassNamesFromComponentObj(json)}>
+					{{${nameStringToVariableString(json.codeContext?.name)}Label}}
+				</ibm-radio>`;
+			}
+		},
+		react: {
+			imports: ['RadioButton'],
+			code: ({ json }) => {
+				return `<RadioButton
+					id="${json.codeContext?.name}"
+					value="${json.value}"
+					checked={${json.defaultChecked}}
```

Comment:
Pretty sure you do not need checked because you are passing defaultChecked & valueChecked in export - Also have a look at the RadioButtonGroup implementation:

https://github.com/carbon-design-system/carbon/blob/9407cb360ceb35ae28fe412e9bcf9af13a9c4650/packages/react/src/components/RadioButtonGroup/RadioButtonGroup.js#L181

That function is printing out the following warning:
```
      /* istanbul ignore if */
      if (typeof radioButton.props.checked !== 'undefined') {
        warning(
          false,
          `Instead of using the checked property on the RadioButton, set
            the defaultSelected property or valueSelected property on the RadioButtonGroup.`
        );
      }
```

Reply:
this works for the wrapper RadioButtonGroup but not <ibm-radio-group>, thats why we used checked
<img width="649" alt="Screen Shot 2022-04-08 at 6 43 53 PM" src="https://user-images.githubusercontent.com/23690711/162543433-07d47a43-190d-43e4-ad01-92094974c60a.png">

<img width="636" alt="Screen Shot 2022-04-08 at 6 47 21 PM" src="https://user-images.githubusercontent.com/23690711/162543634-d4ecef76-fc0d-41f1-a039-a127b4032f3b.png">

Reply:
This is for react, not angular